halpybot/commands/caseutils.py

```
@@ -58,41 +97,555 @@ async def cmd_go(ctx: Context, args: List[str]):
+    )
+    if update:
+        return await ctx.reply(update)
+
+
+@Commands.command("system")
+@Require.permission(Drilled)
+@Require.channel()
+@CaseUtils.gather_case(2)
+async def cmd_system(ctx: Context, args: List[str], case: Case):
+    """
+    Change the system of an active case
+
+    Usage: !rename [board ID] [new system]
+    Aliases: n/a
+    """
+    newsys: str = " ".join(args[1:])
+    newsys = await sys_cleaner(newsys)
+    update = await update_single_elem_case_prep(
+        ctx=ctx, case=case, action="Client System", new_key="system", new_item=newsys
+    )
+    if update:
+        await ctx.reply(update)
+    try:
+        landmark, distance, direction = await checklandmarks(newsys)
+    except (NoResultsEDSM, EDSMConnectionError):
+        return await ctx.reply(
+            f"{newsys} not found in EDSM or unable to connect to EDSM."
+        )
+    return await ctx.reply(
+        f"The closest landmark system is {landmark}, {distance} LY {direction} of {newsys}."
+    )
+
+
+@Commands.command("status")
+@Require.permission(Drilled)
+@Require.channel()
+@CaseUtils.gather_case(2)
+async def cmd_status(ctx: Context, args: List[str], case: Case):
+    """
+    Change the activity status of a case
+
+    Usage: !status [board ID] [new status]
+    Aliases: n/a
+    """
     try:
-        case: Case = await get_case(ctx, args[0])
+        status = Status[args[1].upper()]
+        if status == Status.CLOSED:
+            raise KeyError  # Mock a KeyError. This command can't close a case.
     except KeyError:
-        return await ctx.redirect(f"No case found for {args[0]!r}.")
-    return await ctx.redirect(f"{case}")
+        return await ctx.reply("Invalid case status provided.")
+    update = await update_single_elem_case_prep(
+        ctx=ctx,
+        case=case,
+        action="Case Status",
+        new_key="status",
+        new_item=status,
+        enum=True,
+    )
+    if update:
+        return await ctx.reply(update)
```

this leaves the possibility of a silent failure?

---

Fixed by 7b26b68 in #338. Update will always emit *something*.